src/test/java/org/assertj/core/internal/files/Files_assertHasBinaryContent_Test.java

```diff
@@ -94,13 +108,20 @@ void should_throw_error_wrapping_caught_IOException() throws IOException {
 
   @Test
   void should_fail_if_file_does_not_have_expected_binary_content() throws IOException {
-    BinaryDiffResult diff = new BinaryDiffResult(15, (byte) 0xCA, (byte) 0xFE);
-    when(binaryDiff.diff(actual, expected)).thenReturn(diff);
+    File actual = newFile(tempDir.getAbsolutePath() + "/tmp.txt");
+    byte[] data = "actual".getBytes();
+    try (FileOutputStream myWriter = new FileOutputStream(actual)) {
+      myWriter.write(data, 0, data.length);
+    }
+    byte[] expected = "fake".getBytes();
+
+    BinaryDiff binaryDiff = new BinaryDiff();
+    BinaryDiffResult diff = binaryDiff.diff(actual, expected);
     AssertionInfo info = someInfo();
 
-    Throwable error = catchThrowable(() -> files.assertHasBinaryContent(info, actual, expected));
+    AssertionError error = expectAssertionError(() -> unMockedFiles.assertHasBinaryContent(info, actual, expected));
 
     assertThat(error).isInstanceOf(AssertionError.class);
-    verify(failures).failure(info, shouldHaveBinaryContent(actual, diff));
+    then(error).hasMessage(shouldHaveBinaryContent(actual, diff).create(info.description(), info.representation()));
```

Comment:
Revert this change as the previous code basically does the same thing but in a more concise way

Reply:
this comment applies to other tests too